### PR TITLE
Add basic command line

### PR DESCRIPTION
**Pull request by dkaisr**
Will prompt for input but won't do anything with it yet.
Handles interrupts Ctrl-C, Ctrl-D and Ctrl-\ in interactive mode.

### CODE REVIEW

File: src/repl.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#include "minishell.h"
+
+void	repl(const char *prompt)
+{
+	char	*input;
+
+	while (1)
+	{
+		input = readline(prompt);
```
Comment (dkaisr) [inline src/repl.c:21]: `input` is never freed.

**Review by cuhlig42 — APPROVED** [review]
like it

File: include/minishell.h [hunk 2]
```diff
@@ -6,14 +6,24 @@
...
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/06/22 17:14:49 by dkaiser           #+#    #+#             */
-/*   Updated: 2024/06/24 18:51:30 by dkaiser          ###   ########.fr       */
+/*   Updated: 2024/06/25 13:12:26 by dkaiser          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #ifndef MINISHELL_H
 # define MINISHELL_H
-
```
Comment (dkaisr) [inline include/minishell.h:15]: Why?

File: src/debug_tools.c [hunk 3]
```diff
@@ -6,7 +6,7 @@
 /*   By: dkaiser <dkaiser@student.42heilbronn.de    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/06/24 15:34:14 by dkaiser           #+#    #+#             */
-/*   Updated: 2024/06/25 12:48:20 by dkaiser          ###   ########.fr       */
+/*   Updated: 2024/06/25 13:13:18 by dkaiser          ###   ########.fr       */
```
Comment (dkaisr) [inline src/debug_tools.c:9]: Again: Why?

File: src/main.c [hunk 4]
```diff
@@ -6,13 +6,14 @@
...
+/*   Updated: 2024/06/24 16:13:41 by dkaiser          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "minishell.h"
 
 int	main(void)
 {
-	return (0);
+	init();
```
Comment (dkaisr) [inline src/main.c]: Add error handling if init() fails!

File: src/repl.c (hunk 1 above)
Comment (dkaisr) [reply src/repl.c:21]: Fixed

File: include/minishell.h (hunk 2 above)
Comment (dkaisr) [reply include/minishell.h:15]: No idea. Fixed it.

File: src/debug_tools.c (hunk 3 above)
Comment (dkaisr) [reply src/debug_tools.c:9]: Probably happened while resolving merge conflicts.
Won't fix.

**Review by cuhlig42 — DISMISSED** [review]

File: src/main.c (hunk 4 above)
Comment (dkaisr) [reply src/main.c]: Fixed.

**Review by cuhlig42 — APPROVED** [review]

**Pull request merged**